### PR TITLE
Feature-2532-Adding-Analytics

### PR DESCRIPTION
**Pull request by 11karangm**
Targets #2532

### Description

Created a view to display volunteers that applied today on a chart as bargraph.

### Checklist:

- [ ] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/HR/Http/Controllers/Recruitment/JobController.php [hunk 1]
```diff
@@ -143,4 +148,52 @@ public function storeJobdomain(JobDomainRequest $request)
...
     }
+
+    private function getVerifiedApplicationsCount()
+    {
+        $from = config('hr.verified_application_date.start_date');
+        $currentDate = Carbon::today(config('constants.timezone.indian'));
+
+        return Application::whereBetween('created_at', [$from, $currentDate])
+            ->where('is_verified', 1)->count();
+    }
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/Recruitment/JobController.php:159]: What's the use of this function ?

File: Modules/HR/Http/Controllers/Recruitment/JobController.php [hunk 2]
```diff
@@ -143,4 +148,52 @@ public function storeJobdomain(JobDomainRequest $request)
...
+
+    public function findApplicant()
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/Recruitment/JobController.php:161]: Which applicant we are trying to find here ?

File: Modules/HR/Http/Controllers/Recruitment/JobController.php [hunk 3]
```diff
@@ -143,4 +148,52 @@ public function storeJobdomain(JobDomainRequest $request)
...
+    {
+        $jobs = Job::where('type', 'volunteer')->get('id');
+        //dd($jobs);
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/Recruitment/JobController.php:164]: Let's remove the code if not needed

File: app/Http/Controllers/HR/Volunteers/ReportsController.php [hunk 4]
```diff
@@ -15,4 +19,13 @@ public function index()
...
     }
+    
+    private function getVerifiedApplicationsCount()
+    {
+        $from = config('hr.verified_application_date.start_date');
+        $currentDate = Carbon::today(config('constants.timezone.indian'));
+
+        return Application::whereBetween('created_at', [$from, $currentDate])
+            ->where('is_verified', 1)->count();
+    }
```
Comment (P4NK4J) [inline app/Http/Controllers/HR/Volunteers/ReportsController.php:30]: I see this function was used in the JobController as well ? What's the use and why its repeated ?

File: app/Models/user.php [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class user extends Model
+{
+    use HasFactory;
+}
```
Comment (P4NK4J) [inline app/Models/user.php:11]: Why we need this model file here in app folder ?

File: database/migrations/2022_09_25_131747_add_foreign_key_to_hr_applicants.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('hr_applicants', function (Blueprint $table) {
+            $table->integer('job_id')->unsigned()->nullable();
```
Comment (P4NK4J) [inline database/migrations/2022_09_25_131747_add_foreign_key_to_hr_applicants.php:17]: Why we need to have job_id in applicants table ? job ID should be mapped to application, and applications are mapped to the applicant. I don't understand the use case of this?

File: resources/views/hr/volunteers/reports.blade.php [hunk 7]
```diff
@@ -1,26 +1,31 @@
...
-            <br>
-            @include('hr.volunteers.menu')
-            <br><br>
-        </div>
-        <div class="col-md-12">
-            @include('status', ['errors' => $errors->all()])
-        </div>
-        <div class="col-md-12">
-            <h1>Volunteering Reports</h1>
+    <h1 class="text-center">Reports Vol</h1>
```
Comment (P4NK4J) [inline resources/views/hr/volunteers/reports.blade.php:4]: What's Reports vol ?

File: routes/web.php [hunk 8]
```diff
@@ -35,7 +38,9 @@
     Route::prefix('hr')->namespace('HR')->group(function () {
         Route::prefix('volunteers')->namespace('Volunteers')->group(function () {
             Route::get('reports', 'ReportsController@index')->name('volunteers.reports');
+            //Route::get('reportscount', 'ReportsController@Show')->name('volunteers.dailyapplicationscount');
```
Comment (P4NK4J) [inline routes/web.php:41]: Can we remove this commented code ?

**Review by P4NK4J — CHANGES_REQUESTED** [review]
left some comments, please check

File: database/migrations/2022_09_25_131747_add_foreign_key_to_hr_applicants.php (hunk 6 above)
Comment (11karangm) [reply database/migrations/2022_09_25_131747_add_foreign_key_to_hr_applicants.php:17]: I will delete this file. Made this file when I was a bit confused with the logic. I had deleted that column from my local database.

File: resources/views/hr/volunteers/reports.blade.php (hunk 7 above)
Comment (11karangm) [reply resources/views/hr/volunteers/reports.blade.php:4]: It was the heading of the page , will make it a clearer heading.

File: routes/web.php (hunk 8 above)
Comment (11karangm) [reply routes/web.php:41]: Will delete

File: app/Models/user.php (hunk 5 above)
Comment (11karangm) [reply app/Models/user.php:11]: Sorry created that file by accident when I was working on another laravel project to learn about some eloquent functions.
I will delete that file.

File: Modules/HR/Http/Controllers/Recruitment/JobController.php (hunk 3 above)
Comment (11karangm) [reply Modules/HR/Http/Controllers/Recruitment/JobController.php:164]: I will remove the "dd()" need the job id for performing query.

File: Modules/HR/Http/Controllers/Recruitment/JobController.php (hunk 2 above)
Comment (11karangm) [reply Modules/HR/Http/Controllers/Recruitment/JobController.php:161]: We are trying to find Applications applied by an applicant for volunteer type job.
Or 
In other words we are trying to find the number of Applicants who have applied to become a volunteer and diplay it on the chart.

File: Modules/HR/Http/Controllers/Recruitment/JobController.php (hunk 1 above)
Comment (11karangm) [reply Modules/HR/Http/Controllers/Recruitment/JobController.php:159]: That function was there in the ReportsController in HR module. It gives the count of verified applications.
My chart model was based of the one present in HR module.  It was present in that module. But my guide told to remove it from the volunteers module. I removed the view but forgot to remove the code logic . I will remove it now.

File: Modules/HR/Http/Controllers/Recruitment/JobController.php (hunk 2 above)
Comment (P4NK4J) [reply Modules/HR/Http/Controllers/Recruitment/JobController.php:152]: Why this is written in JobController ? Shouldn't this be a part of applicantController? Also the function name should be `findVolunteerApplicant` or something similar

File: Modules/HR/Http/Controllers/Recruitment/JobController.php [hunk 9]
```diff
@@ -143,4 +148,40 @@ public function storeJobdomain(JobDomainRequest $request)
...
+            ->get();
+
+        $data = [];
+        foreach ($record as $row) {
+            $data['label'][] = (new Carbon($row->date_created_at))->format('M d');
+            $data['data'][] = (int) $row->count;
+            $verifiedApplications = Application::where('is_verified', 1)->whereDate('created_at', '=', date(new Carbon($row->date_created_at)))->count();
+            $data['afterBody'][] = $verifiedApplications;
+        }
+        $data['chartData'] = json_encode($data);
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/Recruitment/JobController.php:180]: This logic should go in applicant service, where you need to send the request data and call service function.

File: resources/views/hr/volunteers/reportresult.blade.php [hunk 10]
```diff
@@ -0,0 +1,40 @@
+@extends('layouts.app')
```
Comment (P4NK4J) [inline resources/views/hr/volunteers/reportresult.blade.php:1]: Can we rename this to `daily-volunteer-application-report.blade.php` if its serving the same purpose ?

File: routes/web.php [hunk 11]
```diff
@@ -165,3 +169,4 @@
     Route::get('user/wishlist-books', 'UserBookController@booksInWishlist');
     Route::get('user/projects', 'UserController@projects');
 });
+Route::get('reportserCard', [JobController::class, 'findApplicant'])->name('volunteers.applicationcount');
```
Comment (P4NK4J) [inline routes/web.php:172]: This route doesn't make sense, `reportserCard` ? Also, why this route is outside auth bracket ?

**Review by P4NK4J — CHANGES_REQUESTED** [review]
Left some comments.

File: Modules/HR/Http/Controllers/Recruitment/JobController.php (hunk 9 above)
Comment (11karangm) [reply Modules/HR/Http/Controllers/Recruitment/JobController.php:180]: I don't understand exactly what you mean.
That is the logic of findVolunteerApplicant().
I am going to put that function in Application Controller according to you instructions. A simmilar logic is there in a simmilar function showReportCard(), and this function is in ReportsController.

**Pull request closed without merging**